src/main/java/se/sundsvall/precheck/service/PreCheckService.java

```diff
@@ -39,13 +40,14 @@ public ResponseEntity<List<PreCheckResponse>> checkPermit(String partyId, String
 
         var citizen = citizenClient.getCitizen(partyId);
         var partyAssets = partyAssetsClient.getPartyAssets(partyId, String.valueOf(Status.ACTIVE));
-        logger.debug("In data resourceNotFound: "+citizen + "    \n\n" + partyAssets);
-         if (resourceNotFound(citizen, partyAssets)) { //TODO: fix always true
-             logger.info("No citizen {} {} or party assets {} {} found for the given partyId: {}", citizen.getStatusCode(), citizen.getBody(), partyAssets.getStatusCode(),partyAssets.getBody(), partyId);
-             throw Problem.valueOf(NO_CONTENT);
-         }
-        logger.debug("All didn't fail?");
-        if (!hasValidMunicipalityId(citizen, municipalityId)) {
+        logger.debug("In data resourceNotFound: " + citizen + "    \n\n" + partyAssets);
+
+        if (resourceNotFound(citizen, partyAssets)) {
+            logger.info(String.format("No citizen {} {} or party assets {} {} found for the given partyId: {}", citizen.getStatusCode(), Optional.ofNullable(citizen.getBody()), partyAssets.getStatusCode(), partyAssets.getBody(), partyId));
+            throw Problem.valueOf(NO_CONTENT, String.format(NO_CONTENT_FOUND, partyId));
```

Comment:
It seems a bit strange to throw an exception with an successful http status code (in the 200-series).
I think NOT_FOUND is more appropriate here.

Reply:
Fixed it